include/my_align_alloc.h

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2000, 2011, Oracle and/or its affiliates. 2016 IBM
+   All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#ifndef _my_aligned_alloc_h
+#define _my_aligned_alloc_h
+
+
+#if defined (_MSC_VER)
+#define ALIGNED_ALLOC(S,A) _aligned_malloc((A), (S))
+#elif defined (_ISOC11_SOURCE)
+#define ALIGNED_ALLOC(S,A)  aligned_alloc((A), (S))
+#elif  _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600
+inline void *XXaligned_alloc(size_t size, size_t align)
+{
+  void *m;
+  errno = posix_memalign(&m, align, size);
+  return errno ? NULL : m;
+}
+#define ALIGNED_ALLOC(S,A)  XXaligned_alloc((S), (A))
+
+#else
+#warn   "No aligned malloc - cache lines may be shared"
```

Comment:
I was hoping so based on the other #warning directives in the code. Thanks for confirming.